data/common/ansible/playbooks/roles/postgresql/tasks/preflight/upgrade.yml

```diff
@@ -94,25 +94,41 @@
         src: "{{ pg_old.pg.data_dir[ansible_os_family] }}/PG_VERSION"
       register: slurp_pg_version
 
-    - name: Ensure that PostgreSQL cluster in version {{ pg_old.pg.version }} is present
+    - name: Assert PostgreSQL cluster in version {{ pg_old.pg.version }} is present
       assert:
         that:
           - slurp_pg_version['content'] | b64decode | trim is version(pg_old.pg.version, '==')
         fail_msg: PostgreSQL cluster in version '{{ pg_old.pg.version }}' not found
         success_msg: PostgreSQL cluster in version '{{ pg_old.pg.version }}' found
         quiet: true
 
-- name: Verify that PostgreSQL {{ pg_old.pg.version }} cluster is healthy
+- name: Assert old PostgreSQL cluster is healthy
   when:
     - not pg_upgrade_already_run
     - pg_manifest.specification.extensions.replication.enabled
   become: true
   become_user: postgres
-  # when there are two primary nodes (after failover) rc is non-zero
-  command: |-
-    {{ pg_old.repmgr.bin_dir[ansible_os_family] }}/repmgr \
-      -f {{ pg_old.repmgr.config_dir[ansible_os_family] }}/repmgr.conf \
-      cluster show --csv
-  changed_when: false
-  delegate_to: "{{ groups.postgresql[0] }}"
-  run_once: true
+  vars:
+    _repmgr_config_file: "{{ pg_old.repmgr.config_dir[ansible_os_family] }}/repmgr.conf"
+  block:
+    - name: Check if PostgreSQL {{ pg_old.pg.version }} cluster is healthy
```

Comment:
Do we need to check cluster status on every node?

Reply:
It's checked only on one node, see line 119 😉